lib/grizzly/commands/table.ex

```diff
@@ -3,680 +3,597 @@ defmodule Grizzly.Commands.Table do
-        handler: {WaitReport, complete_report: :sound_switch_tones_number_report}}},
-      {:sound_switch_tone_info_get,
-       {Commands.SoundSwitchToneInfoGet,
-        handler: {WaitReport, complete_report: :sound_switch_tone_info_report}}},
-      {:sound_switch_configuration_set,
-       {Commands.SoundSwitchConfigurationSet, handler: AckResponse}},
-      {:sound_switch_configuration_get,
-       {Commands.SoundSwitchConfigurationGet,
-        handler: {WaitReport, complete_report: :sound_switch_configuration_report}}},
-      {:sound_switch_tone_play_set, {Commands.SoundSwitchTonePlaySet, handler: AckResponse}},
-      {:sound_switch_tone_play_get,
-       {Commands.SoundSwitchTonePlayGet,
-        handler: {WaitReport, complete_report: :sound_switch_tone_play_report}}},
-
-      # Humidity Control Setpoint
-      {:humidity_control_setpoint_set,
-       {Commands.HumidityControlSetpointSet, handler: AckResponse}},
-      {:humidity_control_setpoint_get,
-       {Commands.HumidityControlSetpointGet,
-        handler: {WaitReport, complete_report: :humidity_control_setpoint_report}}},
-      {:humidity_control_setpoint_supported_get,
-       {Commands.HumidityControlSetpointSupportedGet,
-        handler: {WaitReport, complete_report: :humidity_control_setpoint_supported_report}}},
-      {:humidity_control_setpoint_scale_supported_get,
-       {Commands.HumidityControlSetpointScaleSupportedGet,
-        handler: {WaitReport, complete_report: :humidity_control_setpoint_scale_supported_report}}},
-      {:humidity_control_setpoint_capabilities_get,
-       {Commands.HumidityControlSetpointCapabilitiesGet,
-        handler: {WaitReport, complete_report: :humidity_control_setpoint_capabilities_report}}},
-
-      # Humidity Control Mode
-      {:humidity_control_mode_set, {Commands.HumidityControlModeSet, handler: AckResponse}},
-      {:humidity_control_mode_get,
-       {Commands.HumidityControlModeGet,
-        handler: {WaitReport, complete_report: :humidity_control_mode_report}}},
-      {:humidity_control_mode_supported_get,
-       {Commands.HumidityControlModeSupportedGet,
-        handler: {WaitReport, complete_report: :humidity_control_mode_supported_report}}},
-
-      # Humidity Control Operating State
-      {:humidity_control_operating_state_get,
-       {Commands.HumidityControlOperatingStateGet,
-        handler: {WaitReport, complete_report: :humidity_control_operating_state_report}}}
-    ]
-
-    defmacro __before_compile__(_) do
-      lookup =
-        for {command_name, spec} <- @table do
-          quote location: :keep do
-            def lookup(unquote(command_name)), do: unquote(spec)
-          end
-        end
-
-      quote location: :keep do
-        @doc """
-        Look up the Z-Wave command module and default Grizzly command options via the
-        command name
+  alias Grizzly.CommandHandlers.{AckResponse, AggregateReport, WaitReport}
+  alias Grizzly.ZWave.Commands
+
+  @table %{
```

Comment:
The major change here is the elimination of unnecessary macros. This is now used in a runtime lookup rather than a ton of generated functions (thus the conversion from a list to a table).